src/sage/combinat/sf/sfa.py

```diff
@@ -417,6 +417,26 @@ def is_integral_domain(self, proof=True):
             """
             return self.base_ring().is_integral_domain()
 
+        def fraction_field(self):
+            r"""
+            Return the fraction field of ``self``.
+
+            EXAMPLES::
+
+                sage: s = SymmetricFunctions(QQ).s()
+                sage: s.fraction_field()
+                Fraction Field of Symmetric Functions over Rational Field in the Schur basis
+            """
+            if not self.is_integral_domain():
+                raise TypeError("self must be an integral domain.")
+            if hasattr(self, "__fraction_field") and self.__fraction_field is not None:
+                return self.__fraction_field
+            else:
+                import sage.rings.fraction_field
+                K = sage.rings.fraction_field.FractionField_generic(self)
+                self.__fraction_field = K
+            return self.__fraction_field
```

Comment:
```suggestion
            from sage.rings.fraction_field import FractionField_generic
            return FractionField_generic(self)
```
since we can use the `@cached_method` in the Python class.